homeassistant/components/deconz/light.py

```diff
@@ -158,111 +170,139 @@ def color_mode(self) -> str:
         return color_mode
 
     @property
-    def brightness(self):
+    def brightness(self) -> int | None:
         """Return the brightness of this light between 0..255."""
-        return self._device.brightness
+        if self._device.brightness:
+            return int(self._device.brightness)
```

Comment:
Why is the value being parsed as an integer?
Should it be using casting instead?
Isn't device correctly typed?
Is the upstream library missing the types?